wormhole-connect/src/hooks/useFetchTransactionHistory.ts

```diff
@@ -0,0 +1,333 @@
+      fromChain,
+      tokenKey: tokenConfig.key,
+      receivedTokenKey,
+      receiveAmount: receiveAmountDisplay,
+      relayerFee: {
+        fee: Number(feeAmountDisplay),
+        tokenKey: toChainConfig?.gasToken || '',
+      },
+      tokenAddress: standarizedProperties.tokenAddress,
+      senderTimestamp: sourceChain?.timestamp,
+      receiverTimestamp: targetChain?.timestamp,
+    };
+
+    return txData;
+  };
+
+  // Parser for Portal Token Bridge transactions (appId === PORTAL_TOKEN_BRIDGE)
+  // IMPORTANT: This is where we can add any customizations specific to Token Bridge data
+  // that we have retrieved from WHScan API
+  const parseTokenBridgeTx = (tx: WormholeScanTransaction) => {
+    return parseSingleTx(tx);
+  };
+
+  // Parser for NTT transactions (appId === NATIVE_TOKEN_TRANSFER)
+  // IMPORTANT: This is where we can add any customizations specific to NTT data
+  // that we have retrieved from WHScan API
+  const parseNTTTx = (tx: WormholeScanTransaction) => {
+    return parseSingleTx(tx);
+  };
+
+  // Parser for CCTP transactions (appId === CCTP_WORMHOLE_INTEGRATION)
+  // IMPORTANT: This is where we can add any customizations specific to CCTP data
+  // that we have retrieved from WHScan API
+  const parseCCTPTx = (tx: WormholeScanTransaction) => {
+    return parseSingleTx(tx);
+  };
+
+  const PARSERS = {
+    PORTAL_TOKEN_BRIDGE: parseTokenBridgeTx,
+    NATIVE_TOKEN_TRANSFER: parseNTTTx,
+    CCTP_WORMHOLE_INTEGRATION: parseCCTPTx,
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const parseTransactions = useCallback(
+    (allTxs: Array<WormholeScanTransaction>) => {
+      return allTxs
+        .map((tx) => {
+          // Locate the appIds
+          const appIds: Array<string> =
+            tx.content?.standarizedProperties?.appIds || [];
+
+          for (const appId of appIds) {
+            // Retrieve the parser for an appId
+            const parser = PARSERS[appId];
+
+            // If no parsers specified for the given appIds, we'll skip this transaction
+            if (parser) {
+              return parser(tx);
+            }
+          }
+        })
+        .filter((tx) => !!tx); // Filter out unsupported transactions
+    },
+    [],
+  );
+
+  useEffect(() => {
+    let cancelled = false;
+
+    const headers = new Headers({
+      accept: 'application/json',
+    });
+
+    const fetchTransactions = async () => {
+      setIsFetching(true);
+
+      try {
+        const res = await fetch(
+          `${config.wormholeApi}api/v1/operations?address=${sendingWallet.address}&page=${page}&pageSize=${pageSize}`,
+          { headers },
+        );
+
+        const data = await res.json();
+
+        if (!cancelled) {
+          if (data?.operations?.length > 0) {
+            setTransactions((txs) => {
+              const parsedTxs = parseTransactions(data.operations);
+              if (txs && txs.length > 0) {
+                return txs.concat(parsedTxs);
```

Comment:
It's unlikely, but possible that a user would be infinite scrolling while one of their transactions gets indexed. This would result in every entry in their history being bumped down by 1, and possibly cause you to get a duplicate entry as you page through the history. You could avoid this by de-duplicating them by some kind of unique ID, like the source tx hash.